pkg/orchestrator/assetscanestimationwatcher/watcher.go

```diff
@@ -78,8 +78,8 @@ func (w *Watcher) GetAssetScanEstimations(ctx context.Context) ([]AssetScanEstim
 	logger := log.GetLoggerFromContextOrDiscard(ctx)
 	logger.Debugf("Fetching AssetScanEstimations which need to be reconciled")
 
-	filter := fmt.Sprintf("state/state ne '%s' and state/state ne '%s'",
-		models.AssetScanEstimationStateStateDone, models.AssetScanEstimationStateStateFailed)
+	filter := fmt.Sprintf("(state/state ne '%s' and state/state ne '%s') or (deleteAfter eq null or '%s' > deleteAfter)",
+		models.AssetScanEstimationStateStateDone, models.AssetScanEstimationStateStateFailed, time.Now().String())
```

Comment:
The time here needs to rendered into RFC3339 e.g. `discoveryTime.Format(time.RFC3339)`